Terminal.Gui/View/Layout/DimFill.cs

```diff
@@ -8,19 +8,13 @@ namespace Terminal.Gui;
 ///     This is a low-level API that is typically used internally by the layout system. Use the various static
 ///     methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
 /// </remarks>
-/// <param name="margin">The margin to not fill.</param>
-public class DimFill (int margin) : Dim
+/// <param name="Margin">The margin to not fill.</param>
+public record DimFill (int Margin) : Dim
 {
-    /// <inheritdoc/>
-    public override bool Equals (object? other) { return other is DimFill fill && fill.Margin == Margin; }
-
-    /// <inheritdoc/>
-    public override int GetHashCode () { return Margin.GetHashCode (); }
-
     /// <summary>
     ///     Gets the margin to not fill.
     /// </summary>
-    public int Margin { get; } = margin;
+    public int Margin { get; } = Margin;
```

Comment:
For the _record_ (woo puns!), this is a perfectly valid and complete record definition:

```csharp
// Super basic but has two init-only properties as shown in the constructor.
public record SomeRecord(string Name, SomeType Value);

// Same record type, but declares the interface it naturally supports for free:
public record SomeRecord(string Name, SomeType Value) : IEqualityOperators<SomeRecord<T>, SomeRecord<T>, bool>;
```

And here it is as generic, partial (so you could perhaps put the non-structural code, if any, in a separate file named to nest under it), and with a type parameter constraint that record types implicitly support, so long as you simply declare it, as noted above:

```csharp
public partial record SomeRecord<T> (string Name, T Value) : IEqualityOperators<SomeRecord<T>, SomeRecord<T>, bool> where T : IEqualityOperators<T, T, bool>;
```

You get all the goodies of a record with all three of these. You get like 50 lines of code for the price of one with the third one, and only slightly less with the other two.

Reply:
Also note, since you CANNOT override the equality operators nor the object.Equals method (and therefore _shouldn't_ override the typed `IEquatable<T>.Equals` method either unless you simply pass it through to the generated Equals), it is almost always unnecessary to test equality for record types.

They come with deconstruction built-in, too, like Tuples, so they can also reduce the number of explicit locals that are copied, in consumers of them, too, if you use those, since they provide `out` references to the properties of the record.